### DISCUSSION TITLE
Help onboarding the Appium project

### DISCUSSION THREAD

**Discussion by jlipps** [renovatebot/renovate#10378] [Request Help]
Hi, I'm having a great time working with the [onboarding PR for the Appium project](https://github.com/appium/appium/pull/15480). The one thing I haven't figured out is why the PR details say that the commits will use the `fix` semantic commit type even though I've tried to force it to `chore`. Can anyone advise?

**Comment by HonkingGoose** [comment]
We have some docs about using Semantic Commits: https://docs.renovatebot.com/semantic-commits/

I think this is what you need/want?

> ## Changing the Semantic Commit type
> 
> You can change the Semantic Commit type that Renovate uses.
> 
> Say you want Renovate to use "chore" for every PR, you can add `":semanticCommitTypeAll(chore)"` to your `extends` array.
> PR titles and commit messages start with "chore(deps):"
> 
> Or say you want to use "ci" for every PR, then you would add `":semanticCommitTypeAll(ci)"` to your `extends` array instead.
> PR titles and commit messages start with "ci(deps):"

**Reply by rarkins** [reply]
Yes, I think that should do it. `config:base` includes [:semanticPrefixFixDepsChoreOthers](https://docs.renovatebot.com/presets-default/#semanticprefixfixdepschoreothers) which is an opinionated preset. You could either:

- Add it to `ignoredPresets`, which means that Renovate will revert back to its default behavior which is `chore` semantic commit scope, or
- Add an equally opinionated preset like suggested by @HonkingGoose to override `:semanticPrefixFixDepsChoreOthers`

**Reply by jlipps** [reply]
That worked, thanks @HonkingGoose @rarkins!

**Reply by HonkingGoose** [reply]
You're welcome! I'm glad I could help out! 😄

**Comment by rarkins** [comment]
Really cool to see Appium getting onboarded, too! We're happy to help with your config here any time.

**Reply by jlipps** [reply]
Sounds great, will let you know if we run into anything else

**Reply by jlipps** [reply]
@rarkins can you help me understand why https://github.com/appium/appium/pull/15498 is still trying to pin deps even though I added ":preserveSemverRanges" to my config extensions? I think I also clicked the "rebase" checkbox. Do I need to do anything else to get it to pick up the new config?

**Reply by jlipps** [reply]
or maybe @HonkingGoose knows?

**Reply by HonkingGoose** [reply]
The `:preserveSemVerRanges` preset sets the `rangeStrategy` to `replace` (https://docs.renovatebot.com/presets-default/#preservesemverranges):

```json
{
  "rangeStrategy": "replace"
}
```

I think you actually want the `bump` strategy instead. Please read the docs for `rangeStrategy`: https://docs.renovatebot.com/configuration-options/#rangestrategy

I suspect that removing the `:preserveSemVerRanges` preset, and replacing it with a new `rangeStrategy` key/value pair will get you the desired behavior.
```json
{
  "rangeStrategy": "bump"
}
```

**Reply by rarkins** [reply]
@jlipps although you had `:preserveSemverRanges`, you then had `"rangeStrategy": "auto"` in config which essentially just flipped that setting back again. "auto" essentially means "pin if you think it's a good idea, otherwise replace".


Could you describe what you'd like to see done with ranges? FYI it's not a hill I'll die on, but I'm still to hear any good reason for using ranges for devDependencies in particular.

**Reply by jlipps** [reply]
Thanks for the explanations, this makes sense now. I could be clueless here, but I guess I'm not sure what the value is in pinning. devDeps are just used by devs, and dealing with many many renovate PRs for patch or even minor devDep releases seems like a lot of admin work? Is the advantage supposed to be that you always know you have a green build that won't cause problems for a new dev who pulls down some unknown collection of devDeps that might have something wrong?

**Reply by rarkins** [reply]
With ranges and no lock file, you have no determinism and the project could get broken for all contributors and all PRs at any time.

If you pin/lock, it doesn't mean you have to be non-stop updating, but rather that you have visibility and stability.

I wrote about the topic more here: https://docs.renovatebot.com/dependency-pinning/

**Reply by rarkins** [reply]
BTW with `rangeStrategy=bump` you will get just as many PRs. You may want to go back to `rangeStrategy=replace` if you want to minimize PRs.

**Reply by HonkingGoose** [reply]
> devDeps are just used by devs, and dealing with many many renovate PRs for patch or even minor devDep releases seems like a lot of admin work?

If you have 100% test coverage (or close enough for your comfort) you can setup automerge on at least the minor or patch devDependencies. You could even get Renovate to merge them silently if tests pass, that way the only thing you would see are new update commits on your `main` branch. 😉 

Take a look at our docs on noise reduction, I recommend you read the whole thing: https://docs.renovatebot.com/noise-reduction/ it covers multiple strategies to reduce the noise of updating. 😉

**Reply by jlipps** [reply]
OK these are both very helpful articles. I think you've got me convinced, now I just need to convince the team :-)

**Reply by rarkins** [reply]
It's totally fine to stay how you are though, Renovate will still work happily in either model :)
